web/src/compose_validate.ts

```diff
@@ -281,7 +281,7 @@ export function warn_if_mentioning_unsubscribed_user(
             Number($(user_row).attr("data-user-id")),
         );
 
-        const can_subscribe_other_users = settings_data.user_can_subscribe_other_users();
+        const can_subscribe_other_users = settings_data.can_subscribe_others_to_all_streams();
```

Comment:
`can_subscribe_others_to_all_accessible_streams` might be a good name here too, though it's arguably less important since we shouldn't know about non-accessible streams.